ext/RootsForwardDiffExt.jl

```diff
@@ -0,0 +1,39 @@
+module RootsForwardDiffExt
+
+using Roots
+import Roots: solve
+using ForwardDiff
+import ForwardDiff: Dual, derivative, gradient, partials
+
+
+function solve(ZP::ZeroProblem, M::Roots.AbstractUnivariateZeroMethod, p::Dual{T}; kwargs...) where {T}
+
+    f = ZP.F
+    pᵥ = p.value
+
+    xᵅ = solve(ZP, M, pᵥ; kwargs...)
+
+    fₓ = derivative(x -> f(x, pᵥ), xᵅ)
+    fₚ = derivative(p -> f(xᵅ, p), pᵥ)
```

Comment:
Since you are already dealing with `Dual`s I think it would be more efficient to just pass them around instead of using `derivative`:
```suggestion
    fₓ = ForwardDiff.partials(f(ForwardDiff.Dual{T}(xᵅ, one(xᵅ)), pᵥ), 1)
    fₚ = ForwardDiff.partials(f(xᵅ, p))
```